### PR TITLE
source-mysql: do not use `testcontainers` in integration tests

### PR DESCRIPTION
**Pull request by alafanechere**
## Original problems:
Using `testcontainers` within airbyte-ci and locally has proven to be complicated:
* `testcontainers` used an "old"mysql image which did not have an ARM64 variant. Updating `testcontainers` solves the problem.
* In airbyte-ci `testcontainers` containers and `source-mysql` containers are running on a sidecar docker host which is not the same host as where the test suite is executed. It means that client code (the tests) can reach the `mysql` db container using the sidecar dockerhost hostname, but `source-mysql` can't resolve this host name because it's not running in `host` network mode: it has to reach the `mysql` container using its IP.  It introduces additional networking challenges in intregration tests, when the connector container has to communicate with the db connector.  The difference with local execution of the integration test is that in local execution the docker host == localhost.  
* `testcontainers` spawns a lot of ephemeral database containers. It is resource intensive and makes running local / airbyte-ci hard.
* The `source-mysql` integration test suite has a lot of duplicated code, which makes it hard to understand the test flow and context.

## Suggested approach
* Discard the use of `testcontainers` for integration test: use a real remote DB provisionned on CloudSQL (GCP)

## Leave the the campground cleaner than you found it
* Refactor the test suite to reduce the duplicated code
* Improve the gradle caching logic in airbyte-ci to speed up gradle tasks:
  * Enable local build cache when running within airbyte-ci
  * Make gradle container persist their `/root/.gradle/` directory to a cache volume

## Open questions for java source developers
* Shall we continue to consider the `JdbcSourceAcceptanceTest` test suite as part of the unit test? Are you running this test suite locally when iterating on a connector? This test suite uses `testcontainers`. It has no networking issue while running in airbyte-ci because these tests are not launching a connector as a container but use the `MySqlSource` object that can directly connect . But as, I mentioned above, using `testcontainers` adds a lot of overhead to test execution, can we use the remote test instance?

## Current blockers
**The java integration tests are hanging. When running `airbyte-ci connectors --name=source-mysql test` everything runs fine until the integration tests. Then the integration tests starts but the execution of `./gradlew --no-daemon --scan --build-cache --debug :airbyte-integrations:connectors:source-mysql:integrationTest -x airbyteDocker` is not returning any progress while eating a ton of CPU. **

## TODO
* Fix failing state related tests 
* Propagate the same approach to `source-mysql-strict-encrypt`
* Make the `gradle-cache` volume persisted by Dagger's magicache to benefit from it in the CI.

### CODE REVIEW

File: airbyte-integrations/connectors/source-mysql-strict-encrypt/src/test-integration/java/io/airbyte/integrations/source/mysql_strict_encrypt/AbstractMySqlSslCertificateStrictEncryptSourceAcceptanceTest.java [hunk 1]
```diff
@@ -37,8 +39,8 @@ protected void setupEnvironment(final TestDestinationEnv environment) throws Exc
         .put("method", "STANDARD")
         .build());
     config = Jsons.jsonNode(ImmutableMap.builder()
-        .put(JdbcUtils.HOST_KEY, innerContainerAddress.left)
-        .put(JdbcUtils.PORT_KEY, innerContainerAddress.right)
+        .put(JdbcUtils.HOST_KEY, HostPortResolver.resolveHost(container))
+        .put(JdbcUtils.PORT_KEY, HostPortResolver.resolvePort(container))
```
Comment (rodireich) [inline airbyte-integrations/connectors/source-mysql-strict-encrypt/src/test-integration/java/io/airbyte/integrations/source/mysql_strict_encrypt/AbstractMySqlSslCertificateStrictEncryptSourceAcceptanceTest.java:43]: Does this still work when running directly on macOS with gradle `:integrationTestJava`?

**Comment by alafanechere** [comment]
Closing as this is was fixed in other PRs

**Pull request closed without merging**